msticpy/nbtools/morph_charts.py

```diff
@@ -0,0 +1,156 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""Morph Charts  class."""
+import json
+from pathlib import Path
+import yaml
+from IPython.display import IFrame
+import pandas as pd
+from .utility import MsticpyException
+
+from .._version import VERSION
+
+__version__ = VERSION
+__author__ = "Pete Bryan"
+
+_CHART_FOLDER = "morph_charts"
+
+
+class MorphCharts:
+    """Create Morph Charts package data and render Morph Charts site."""
+
+    def __init__(self):
+        """Create object and populate charts container."""
+        self.charts = _get_charts(_CHART_FOLDER)
+
+    def display(self, data: pd.DataFrame, chart_name: str) -> IFrame:
+        """
+        Prepare package data and display MorphChart in an IFrame.
+
+        Parameters
+        ----------
+        data: pd.DataFrame:
+            A DataFrame of data for the morphchart to plot.
+
+        chart_name: str:
+            The name of the Morph Chart to plot.
+
+        """
+        # Check input data is correct format and that the chart being requested exists
+        if not isinstance(data, pd.DataFrame):
+            raise MsticpyException("Data provided must be in pandas.DataFrame format")
+
+        try:
```

Comment:
You could just use something like self.charts.get(chart_name) or if chart_name not in self.charts: